src/layout.rs

```diff
@@ -790,12 +790,183 @@ impl Element {
+///
+/// This macro simplifies the process of creating various constraints.
+#[macro_export]
+macro_rules! constraints {
+    ([ $( $constraint:tt )+ ]) => {
+        constraints!(@parse () $($constraint)+)
+    };
+    // Internal parsing rule for constraints
+    // This rule checks if `,` exists after the `head` token
+    (@parse ($($acc:tt)*) $head:tt , $($tail:tt)*) => {
+        // Combines the head constraint with the tail constraints into a vector
+        std::iter::once(
+            constraints!(@process ($($acc)* $head)) // -> Constraint
+        ).chain(
+            constraints!(@parse () $($tail)*).into_iter() // -> Iterator with type Constraint
+        )
+    };
+    // If there is no `,` then accumulate `next` token into existing `acc` tokens
+    // and return `Iterator`
+    (@parse ($($acc:tt)*) $next:tt $($tail:tt)*) => {
+        constraints!(@parse ($($acc)* $next) $($tail)*)
+    };
+    // At the end there will a set of tokens left after the last `,`
+    // Process that as a `Constraint`
+    (@parse ($($acc:tt)*)) => {
+        vec![constraints!(@process ($($acc)*))]
+    };
+    // Process different types of constraints into a `Constraint`
+    (@process (== $token1:tt / $token2:tt)) => {
+        // Ratio constraint
+        {
+        let t1: u32 = $token1;
+        let t2: u32 = $token2;
+        $crate::prelude::Constraint::Ratio(t1, t2)
+        }
+    };
+    (@process (== $token:tt %)) => {
+        // Percentage constraint
+        $crate::prelude::Constraint::Percentage($token)
+    };
+    (@process (>= $token:tt)) => {
+        // Minimum size constraint
+        $crate::prelude::Constraint::Min($token)
+    };
+    (@process (<= $token:tt)) => {
+        // Maximum size constraint
+        $crate::prelude::Constraint::Max($token)
+    };
+    (@process (== $token:tt)) => {
+        // Fixed size constraint
+        $crate::prelude::Constraint::Length($token)
+    };
+}
+
+/// This macro creates a layout with specified constraints and direction.
+///
+/// # Syntax
+///
+/// The macro supports three main forms:
+/// - `layout!([$( $constraint:tt )+])`: Defines a default layout (vertical) with constraints.
+/// - `layout!([$( $constraint:tt )+], orient = h)`: Defines a horizontal layout with constraints.
+/// - `layout!([$( $constraint:tt )+], orient = v)`: Defines a vertical layout with constraints.
+///
+/// Constraints are defined using a specific syntax:
+/// - `== $token:tt / $token2:tt`: Sets a ratio constraint between two tokens.
+/// - `== $token:tt %`: Sets a percentage constraint for the token.
+/// - `>= $token:tt`: Sets a minimum size constraint for the token.
+/// - `<= $token:tt`: Sets a maximum size constraint for the token.
+/// - `== $token:tt`: Sets a fixed size constraint for the token.
+///
+/// # Examples
+///
+/// ```
+/// // Vertical layout with fixed size and percentage constraints
+/// use ratatui::layout;
+/// use ratatui::constraints;
+/// layout!([== 50, == 30%]);
+/// ```
```

Comment:
Oddly though, you have to import both `layout` and `constraints` to use `layout!`.

Doing this was throwing an error for the doctests:

```rust
use ratatui::layout;
layout!([==50, ==30%]);
```

Reply:
Resolved by adding a prefix everywhere, i.e. `$crate::constraints!`